pilot/cmd/istioctl/controlplane/cmd.go

```diff
@@ -0,0 +1,149 @@
+			"features requested by the --feature flag. If no features are provided, we create deployments for the " +
+			"default control plane: Pilot, Mixer, CA, an initializer for automatic side-car injection, and Ingress " +
+			"Proxies, with mTLS enabled.",
+		Example: `istioctl control-plane --features routing,policy,initializer -o helm`,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := install.setFeatures(*features); err != nil {
+				return err
+			}
+			install.Namespace = *istioNamespaceFlag
+			switch strings.ToLower(out) {
+			case "helm":
+				_, err := fmt.Fprint(os.Stdout, valuesFromInstallation(install))
+				return err
+			case "yaml":
+				rendered, err := yamlFromInstallation(install, helmChartLocation)
+				if err != nil {
+					return err
+				}
+				_, err = fmt.Fprint(os.Stdout, rendered)
+				return err
+			default:
+				return fmt.Errorf("unsupported output %q", out)
+			}
+		},
+	}
+
+	features = cmd.PersistentFlags().StringArrayP("features", "f", []string{},
+		`List of Istio features to enable. Accepts any combination of "mtls", "telemetry", "routing", "ingress", "policy", "initializer".`)
+	cmd.PersistentFlags().StringVar(&install.Hub, "hub", "gcr.io/istio-testing", "The container registry to pull Istio images from")
+	cmd.PersistentFlags().StringVar(&install.MixerTag, "mixer-tag", "latest", "The tag to use to pull the `mixer` container")
+	cmd.PersistentFlags().StringVar(&install.PilotTag, "pilot-tag", "latest", "The tag to use to pull the `pilot-discovery` container")
+	cmd.PersistentFlags().StringVar(&install.CaTag, "ca-tag", "latest", "The tag to use to pull the `ca` container")
+	cmd.PersistentFlags().StringVar(&install.ProxyTag, "proxy-tag", "latest", "The tag to use to pull the `proxy` container")
+	cmd.PersistentFlags().BoolVar(&install.Debug, "debug", false, "If true, uses debug images instead of release images")
+	cmd.PersistentFlags().Uint16Var(&install.NodePort, "ingress-node-port", 0,
+		"If provided, Istio ingress proxies will run as a NodePort service mapped to the port provided by this flag. "+
+			"Note that this flag is ignored unless the \"ingress\" feature flag is provided too.")
+	cmd.PersistentFlags().StringVarP(&out, "out", "o", "helm", `Output format. Acceptable values are:
+					"helm": produces contents of values.yaml
+					"yaml": produces Kubernetes deployments`)
+
+	// TODO: figure out how we want to package up the charts with the binary to make this easy
+	cmd.PersistentFlags().StringVar(&helmChartLocation, "helm-chart-dir", ".",
+		"The directory to find the helm charts used to render Istio deployments. -o yaml uses these to render the helm chart locally.")
+
+	_ = cmd.PersistentFlags().MarkHidden("hub")
+	_ = cmd.PersistentFlags().MarkHidden("mixer-tag")
+	_ = cmd.PersistentFlags().MarkHidden("pilot-tag")
+	_ = cmd.PersistentFlags().MarkHidden("ca-tag")
+	_ = cmd.PersistentFlags().MarkHidden("proxy-tag")
+	return cmd
+}
+
+type installation struct {
+	Mixer       bool
```

Comment:
enable_mixer bool
enable_pilot bool
...

Reply:
or better change them to  disable_xxx so that default values are false.

Reply:
`Enable*` is not super go idiomatic: their usage is pretty obvious from context, I don't think the extra bits in the name are necessary. We can update them down the line if it becomes non-obvious due to adding other options.